tests/engine/test_engine.py

```diff
@@ -29,9 +31,9 @@ def test_set_subnet_tag(default_subnet, subnet_arg, expected_subnet, monkeypatch
     monkeypatch.setattr(yapapi.engine, "DEFAULT_SUBNET", default_subnet)
 
     if subnet_arg is not None:
-        golem = Golem(budget=1.0, subnet_tag=subnet_arg)
+        golem = GolemFactory(budget=1.0, subnet_tag=subnet_arg)
```

Comment:
This make it so we don't need to pass app_key everywhere. Before it was done by mocking `rest.Configuration`.

Instead of using `GolemFactory` I can:

1. Make a fixture that will set YAGNA_APPKEY
2. Pass ApiConfigFactory in every `Golem(..., api_config=ApiConfigFactory())`
3. Mock `ApiConfig` like it was done with `rest.Configuration` with `monkeypatch.setattr(yapapi.rest, "Configuration", Mock)`